android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java

```diff
@@ -138,6 +138,38 @@ protected Void doInBackground(final Void ... params) {
         myAsyncTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
     }
 
+
+    /*
+     * Returns all contacts matching a phone number
+     */
+    @ReactMethod
+    public void getContactsByPhoneNumber(final String phoneNumber, final Callback callback) {
+        getAllContactsByPhoneNumber(phoneNumber, callback);
+    }
+
+    /**
+     * Retrieves contacts matching a phone number.
+     * Uses raw URI when <code>rawUri</code> is <code>true</code>, makes assets copy otherwise.
+     *
+     * @param phoneNumber phone number to match
+     * @param callback user provided callback to run at completion
+     */
+    private void getAllContactsByPhoneNumber(final String phoneNumber, final Callback callback) {
+        AsyncTask<Void,Void,Void> myAsyncTask = new AsyncTask<Void,Void,Void>() {
```

Comment:
This function body should be inlined into `getContactsByPhoneNumber` on line 147. I think `getAllContactsMatchingString` was split into two function due to misunderstanding what was going on in `getAll`,`getAllWithoutPhotos` and `getAllContacts`.

@hatemalimam is it necessary for `getAllContactsByPhoneNumber` to be two functions? one decorated by `@ReactMethod` and one not decorated.

Reply:
In fact I see no reason to have them split like that, I was just following the one done in `getAllContactsMatchingString`, as it implies some convention.

I'll submit a refactor PR.

Reply:
awesome! Things like this happen when multiple people work on a codebase, you don't remember why something was done the way it was. And the comments don't explain either.